plugin/pkg/shoot/validator/admission.go

```diff
@@ -134,20 +135,60 @@ func (v *ValidateShoot) Admit(a admission.Attributes, o admission.ObjectInterfac
 		return nil
 	}
 
-	// Ignore updates to shoot status or other subresources
-	if a.GetSubresource() != "" {
-		return nil
-	}
-
 	shoot, ok := a.GetObject().(*garden.Shoot)
 	if !ok {
 		return apierrors.NewInternalError(errors.New("could not convert resource into Shoot object"))
 	}
 
+	var oldShoot *garden.Shoot
+
+	switch a.GetOperation() {
+	case admission.Create:
+		// We only want to validate fields in the Shoot against the CloudProfile/Seed constraints which have changed.
+		// On CREATE operations we just use an empty Shoot object, forcing the validator functions to always validate.
+		// On UPDATE operations we fetch the current Shoot object.
+		oldShoot = &garden.Shoot{
+			Spec: garden.ShootSpec{
+				Cloud: garden.Cloud{
+					AWS: &garden.AWSCloud{
+						MachineImage: &garden.ShootMachineImage{},
+					},
+					Azure: &garden.AzureCloud{
+						MachineImage: &garden.ShootMachineImage{},
+					},
+					GCP: &garden.GCPCloud{
+						MachineImage: &garden.ShootMachineImage{},
+					},
+					Packet: &garden.PacketCloud{
+						MachineImage: &garden.ShootMachineImage{},
+					},
+					OpenStack: &garden.OpenStackCloud{
+						MachineImage: &garden.ShootMachineImage{},
+					},
+					Alicloud: &garden.Alicloud{
+						MachineImage: &garden.ShootMachineImage{},
+					},
+				},
+			},
+		}
+	case admission.Update:
+		var ok bool
+		oldShoot, ok = a.GetOldObject().(*garden.Shoot)
+		if !ok {
+			return apierrors.NewInternalError(errors.New("could not convert old resource into Shoot object"))
+		}
+	}
+
+	// Ignore changes to anything but the shoot spec
```

Comment:
Move this up in the `admission.Update` such that it's clear that it only happens for update requests.